src/ui/simulator/toolbox/components/datagrid/renderer/column.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+** Copyright 2007-2018 RTE
+** Authors: Antares_Simulator Team
+**
+** This file is part of Antares_Simulator.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+
+#include "column.h"
+#include "cell.h"
+
+
+namespace Antares
+{
+namespace Component
+{
+namespace Datagrid
+{
+namespace Renderer
+{
+
+// -------------------
+// Base column class
+// -------------------
+Column::Column()
+{}
+
+cell* Column::getLine(int y) const
+{
+	return cells_[y].get();
+}
+
+int Column::getNumberOfLines() const
+{
+	return cells_.size();
+}
+
+// -------------------- -
+// Classic column
+// ---------------------
+classicColumn::classicColumn(TimeSeries ts) : Column()
+{
+	tsKind_ = ts;
+
+	cells_.push_back(make_unique<blankCell>());
```

Comment:
**make_unique** is unknown on Linux, compiler does not know it.
Could be solved (or not) in cmake files with : 
`set(CMAKE_CXX_STANDARD 17) `or `set(CMAKE_CXX_STANDARD 14)`

Reply:
No need for `std::unique_ptr` here, since the memory is handled by objects. We can just free the memory at object destruction.